app/helpers/application_helper.rb

```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+  def full_title page_title
+    base_title = t "layouts.application.suffix_title"
+    page_title.empty? ? base_title : page_title + " | " + base_title
```

Comment:
so sánh nil?, empty?, blank?, present?
trường hợp này nên dùng cái nào

Reply:
**nil?** chỉ để check đối tượng có nil hay không nên **chưa chặt**. (không phải nil chưa chắc đã là chuỗi rỗng)
**empty?** sẽ check chuỗi dựa trên kiểm tra trên độ dài của chuỗi nhưng đối tượng gọi method phải không nil. Đồng thời với chuỗi toàn khoảng trắng thì method vẫn trả về false (tức chuỗi không rỗng). **→chưa chặt và có thể gặp lỗi.**
**blank?** sẽ check chuỗi dựa trên kiểm tra trên độ dài của chuỗi và thêm cả dãy các khoảng trắng. Đồng thời lại không phải kiểm tra nil. **Theo sũy nghĩ của em, chặt và nên dùng**

**present?** ngược của **blank?** nên phải đảo 2 vế xử lý phía sau.